scenarios/getting_started/python/getting_started.py

```diff
@@ -9,56 +16,89 @@ def on_connect(client, _userdata, _flags, rc):
+            transport="tcp",
+        )
+    if 'MQTT_USERNAME' in connection_settings:
+        mqtt_client.username_pw_set(
+                username=connection_settings['MQTT_USERNAME'],
+                password=connection_settings['MQTT_PASSWORD'] if 'MQTT_PASSWORD' in connection_settings else None
+        )
+    if connection_settings['MQTT_USE_TLS']:
+        context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+
+        if connection_settings['MQTT_CERT_FILE']:
+            context.load_cert_chain(
+                    certfile=connection_settings['MQTT_CERT_FILE'],
+                    keyfile=connection_settings['MQTT_KEY_FILE'],
+                    password=connection_settings['MQTT_KEY_FILE_PASSWORD']
+            )
+
+        if "MQTT_CA_FILE" in connection_settings:
+            context.load_verify_locations(
+                    cafile=connection_settings['MQTT_CA_FILE'],
+            )
+        elif "MQTT_CA_PATH" in connection_settings:
+                context.load_verify_locations(
+                    capath=connection_settings['MQTT_CA_PATH']
+            )
+        else:
+            context.load_default_certs()
+
+        mqtt_client.tls_set_context(context)
+    return mqtt_client
+
 parser = ArgumentParser()
 parser.add_argument("--env-file", help="path to the .env file to use")
 args = parser.parse_args()
 
-settings = scenario_settings.get_settings(args.env_file)
+logging.basicConfig(level=logging.DEBUG)
 
-if settings["MQTT_CLEAN_SESSION"] == False:
+connection_settings = cs.get_connection_settings(args.env_file)
+if not connection_settings["MQTT_CLEAN_SESSION"]:
     raise ValueError("This sample does not support connecting with existing sessions")
 
+# INITIALIZE
 print("Initializing Paho MQTT client")
-client = mqtt.Client(settings["MQTT_CLIENT_ID"])
-
-if settings["MQTT_USERNAME"] is not None:
-    client.username_pw_set(settings["MQTT_USERNAME"], settings["MQTT_PASSWORD"])
+client_id = connection_settings["MQTT_CLIENT_ID"]
+mqtt_client = create_mqtt_client(client_id, connection_settings)
 
-if settings["MQTT_USE_TLS"]:
-    client.tls_set(
-        ca_certs=settings["MQTT_CA_FILE"],
-        certfile=settings["MQTT_CERT_FILE"],
-        keyfile=settings["MQTT_KEY_FILE"],
-        keyfile_password=settings["MQTT_KEY_FILE_PASSWORD"],
-    )
+# ATTACH HANDLERS
+mqtt_client.on_connect = on_connect
+mqtt_client.on_message = on_message
+mqtt_client.on_publish = on_publish
+mqtt_client.on_subscribe = on_subscribe
+mqtt_client.on_message = on_message
 
-client.on_connect = on_connect
-client.on_message = on_message
-client.on_publish = on_publish
-client.on_subscribe = on_subscribe
+# CONNECT
+print("{}: Starting connection".format(client_id))
+hostname = connection_settings['MQTT_HOST_NAME']
+port = connection_settings['MQTT_TCP_PORT']
+keepalive = connection_settings["MQTT_KEEP_ALIVE_IN_SECONDS"]
+mqtt_client.connect(hostname, port,keepalive)
+print("Starting network loop")
+mqtt_client.loop_forever()
 
```

Comment:
subscribes and publishes should happen here

Reply:
all callback set conditions now